### PR TITLE
Resolve some issues related with coredns

### PR DESCRIPTION
**Pull request by Hong-Chang**

### CODE REVIEW

File: rdma_srv/src/rdma_agent.rs [hunk 1]
```diff
@@ -442,10 +442,10 @@ impl RDMAAgent {
             RDMAReqMsg::RDMAConnect(msg) => {
                 let mut msgCloned = msg.clone();
 
-                match RDMA_CTLINFO.IsService(msgCloned.dstIpAddr, &msgCloned.dstPort) {
+                match RDMA_CTLINFO.IsService(msgCloned.dstIpAddr, String::from("TCP"), &msgCloned.dstPort) {
```
Comment (qingqu) [inline rdma_srv/src/rdma_agent.rs:445]: use constant

File: rdma_srv/src/rdma_agent.rs [hunk 2]
```diff
@@ -442,10 +442,10 @@ impl RDMAAgent {
...
                     None => {}
                     Some(ipWithPort) => {
-                        println!("RDMAConnect: The traffic is connecting to a service. Change the connection to {:?}", ipWithPort);
+                        println!("RDMAConnect: The traffic {} is connecting to a service. Change the connection to {:?}", msgCloned.dstIpAddr, ipWithPort);
```
Comment (qingqu) [inline rdma_srv/src/rdma_agent.rs:448]: comment

File: rdma_srv/src/rdma_agent.rs [hunk 3]
```diff
@@ -626,6 +626,10 @@ impl RDMAAgent {
                 let vpcId;
                 let ipAddr;
                 if RDMA_CTLINFO.isK8s {
+                    while self.GetVpcId() == 0 {
+                        println!("Waiting for control plane to get vpcId");
+                        sleep(Duration::from_millis(1000));
```
Comment (qingqu) [inline rdma_srv/src/rdma_agent.rs:631]: +TODO: need find a workaround instead of loop here.

File: rdma_srv/src/rdma_ctrlconn.rs [hunk 4]
```diff
@@ -251,26 +251,25 @@ impl CtrlInfo {
         None
     }
 
-    pub fn IsService(&self, ip: u32, port: &u16) -> Option<IpWithPort> {
+    pub fn IsService(&self, ip: u32, protocal: String, port: &u16) -> Option<IpWithPort> {
```
Comment (qingqu) [inline rdma_srv/src/rdma_ctrlconn.rs:254]: protocol

File: rdma_srv/src/rdma_ctrlconn.rs [hunk 5]
```diff
@@ -251,26 +251,25 @@ impl CtrlInfo {
...
         let services = self.services.lock();
         if services.contains_key(&ip) {
             for p in services[&ip].ports.iter() {
-                if p.port == *port {
+                if p.protocal == protocal && p.port == *port {
```
Comment (qingqu) [inline rdma_srv/src/rdma_ctrlconn.rs:258]: same

Comment (Hong-Chang) [reply rdma_srv/src/rdma_ctrlconn.rs:258]: Fixed

File: rdma_srv/src/rdma_ctrlconn.rs (hunk 4 above)
Comment (Hong-Chang) [reply rdma_srv/src/rdma_ctrlconn.rs:254]: Fixed

File: rdma_srv/src/rdma_agent.rs (hunk 3 above)
Comment (Hong-Chang) [reply rdma_srv/src/rdma_agent.rs:631]: Added

File: rdma_srv/src/rdma_agent.rs (hunk 2 above)
Comment (Hong-Chang) [reply rdma_srv/src/rdma_agent.rs:448]: Commented

**Pull request merged**